salt/modules/cp.py

```diff
@@ -900,7 +903,7 @@ def push_dir(path, glob=None, upload_path=None):
         return push(path, upload_path=upload_path)
     else:
         filelist = []
-        for root, _, files in os.walk(path):
+        for root, _, files in os.walk(ur''.join(path)):
```

Comment:
Honestly, it would be best if we didn't try to use a str.join as a dirty hack, so the optimal solution doesn't even use this.